fhir-query-helper-r4/src/test/java/io/elimu/a2d2/cds/fhir/helper/test/FhirTest.java

```diff
@@ -1,15 +1,88 @@
 package io.elimu.a2d2.cds.fhir.helper.test;
 
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
 import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.InstantType;
+import org.hl7.fhir.r4.model.Meta;
+import org.hl7.fhir.r4.model.StructureDefinition;
 import org.junit.Assert;
 import org.junit.Assume;
+import org.junit.Before;
 import org.junit.Test;
 
+import io.elimu.a2d2.cds.fhir.cache.CacheUtil;
 import io.elimu.a2d2.cds.fhir.helper.FhirResponse;
+import io.elimu.a2d2.cds.fhir.helper.QueryBuilder;
 import io.elimu.a2d2.cds.fhir.helper.r4.QueryingServerHelper;
 
 public class FhirTest {
 
+	public static QueryingServerHelper queryingServerHelper = spy(QueryingServerHelper.class);
+	
+	private static final String user_1 = "user_22";
+	private static final String pass_1 = "pass_23";
+	private static final String fhirUrl_1 = "https://fhirUrl_1";
+	private static final String resourceQuery_1 = "https://fhirUrl_1/MedicationAdministration?_count=1&_sort=-_lastUpdated";
+	private static final String responseStatusInfo_1 = "responseStatusInfo - response first user_1";
+	FhirResponse<List<IBaseResource>> retval_1 = null;
+
+	private static final String user_2 = "user2";
+	private static final String pass_2 = "pass2";
+	private static final String fhirUrl_2 = "https://fhirUrl_2";
+	private static final String resourceQuery_2 = "https://fhirUrl_2/Observation?_count=2&_sort=-date";
+	private static final String responseStatusInfo_2 = "responseStatusInfo - response first user_2";
+	FhirResponse<List<IBaseResource>> retval_2 = null;
+
+
+	@Before
+	public void setup() {
+		List<IBaseResource> iBaseList = new ArrayList<IBaseResource>();
+		StructureDefinition iBase = new StructureDefinition();
+		//iBase.setId(new IdDt("testId"));
+		iBase.setUrl("test_url");
+		Meta meta = new Meta();
+		InstantType instantType = new InstantType(Calendar.getInstance());
+		meta.setLastUpdatedElement(instantType);
+		iBase.setMeta(meta);
+		iBaseList.add(iBase);
+
+		retval_1 = new FhirResponse<>(iBaseList, 200, responseStatusInfo_1);
+		retval_2 = new FhirResponse<>(iBaseList, 200, responseStatusInfo_2);
+
+		doReturn(retval_1).when(queryingServerHelper).queryServer(resourceQuery_1, false);
+		doReturn(retval_2).when(queryingServerHelper).queryServer(resourceQuery_2, true);
+	}
+	
+	@Test
+	public void testQueryWithQueryHelper() {
+		clearCache();
+		//user1 url1
+		queryingServerHelper.setFhirUrl(fhirUrl_1);
+		queryingServerHelper.addAuthentication(CacheUtil.BASIC, user_1, pass_1);
+		FhirResponse<List<IBaseResource>> retval = queryingServerHelper.query(
```

Comment:
Nice!

Reply:
thanks, sorry, I need you to review it again because I added an "id" field to the QueryBuilder. It will help to use this from multiple rules (I will create a wiki page about that tomorrow morning. I promised Sofia as much :) )